### PR TITLE
Messaging tests

### PR DESCRIPTION
**Pull request by swarna04**
## Description

- Added MessagingProposition, MessagingPropositionInteraction, MessagingPropositionItemTests.
- Fixed implementation to not allow empty `scopeDetails` in MessagingProposition.
- Fixed implementation to not allow empty `itemData` in MessagingPropositionItem and fixed corresponding tests.
- Removed unused files.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPMessaging/Sources/MessagingPropositionItem.swift [hunk 1]
```diff
@@ -56,8 +56,11 @@ public class MessagingPropositionItem: NSObject, Codable {
 
         itemId = try container.decode(String.self, forKey: .id)
         schema = try SchemaType(from: container.decode(String.self, forKey: .schema))
-        let codableItemData = try? container.decode([String: AnyCodable].self, forKey: .data)
-        itemData = AnyCodable.toAnyDictionary(dictionary: codableItemData)
+        let codableItemData = try container.decode([String: AnyCodable].self, forKey: .data)
+        itemData = AnyCodable.toAnyDictionary(dictionary: codableItemData) ?? [:]
+        guard !itemData.isEmpty else {
+            throw DecodingError.dataCorruptedError(forKey: CodingKeys.data, in: container, debugDescription: "Item data is corrupted and cannot be decoded.")
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/MessagingPropositionItem.swift:62]: debug description is a bit misleading here i think - the data isn't corrupt, it's just empty which is unexpected.

File: AEPMessaging/Sources/MessagingProposition.swift [hunk 2]
```diff
@@ -54,8 +54,11 @@ public class MessagingProposition: NSObject, Codable {
 
         uniqueId = try container.decode(String.self, forKey: .id)
         scope = try container.decode(String.self, forKey: .scope)
-        let anyCodableDict = try? container.decode([String: AnyCodable].self, forKey: .scopeDetails)
-        scopeDetails = AnyCodable.toAnyDictionary(dictionary: anyCodableDict) ?? [:]
+        let codableScopeDetails = try? container.decode([String: AnyCodable].self, forKey: .scopeDetails)
+        scopeDetails = AnyCodable.toAnyDictionary(dictionary: codableScopeDetails) ?? [:]
+        guard !scopeDetails.isEmpty else {
+            throw DecodingError.dataCorruptedError(forKey: CodingKeys.scopeDetails, in: container, debugDescription: "Scope details is corrupted and cannot be decoded.")
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/MessagingProposition.swift:60]: same here about the debug description

**Review by sbenedicadb — APPROVED** [review]
approved - a couple nit comments

File: AEPMessaging/Sources/MessagingPropositionItem.swift (hunk 1 above)
Comment (swarna04) [reply AEPMessaging/Sources/MessagingPropositionItem.swift]: As discussed offline, I've updated the implementation to allow empty data for holdout population. Also added a test for the same.

File: AEPMessaging/Sources/MessagingProposition.swift (hunk 2 above)
Comment (swarna04) [reply AEPMessaging/Sources/MessagingProposition.swift]: N/A, updated the pr as described above.

Comment (swarna04) [reply AEPMessaging/Sources/MessagingProposition.swift:60]: I'll keep this check as discussed.

**Pull request merged**